src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisDataFetcher.java

```diff
@@ -113,13 +125,18 @@ public GetRecordsResult getResult() {
 
         @Override
         public GetRecordsResult accept() {
+            if (!isValidResult(result)) {
+                throw new SdkClientException("GetRecordsResult is not valid. NextShardIterator: " + result.getNextShardIterator()
+                                                     + ". ChildShards: " + result.getChildShards());
```

Comment:
Do we want to include more information about why the result is invalid? Customers might be confused about the relationship between nextShardIterator and childShards if they don't look at the code. We could also move the throwable to isValidResult() so that we can customize the error message based on which combination of nextShardIterator and childShards we receive from the service.